flutter/lib/src/view_hierarchy/sentry_tree_walker.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:flutter/widgets.dart';
+
+import '../../sentry_flutter.dart';
+import '../widget_utils.dart';
+
+// adapted from https://github.com/ueman/sentry-dart-tools/blob/8e41418c0f2c62dc88292cf32a4f22e79112b744/sentry_flutter_plus/lib/src/integrations/tree_walker_integration.dart
+
+class _TreeWalker {
+  static const _privateDelimiter = '_';
+
+  _TreeWalker(this.rootElement);
+
+  final Element rootElement;
+
+  ValueChanged<Element> _visitor(
+      SentryViewHierarchyElement parentSentryElement) {
+    return (Element element) {
+      final sentryElement = _toSentryViewHierarchyElement(element);
+
+      var privateElement = false;
+      // when obfuscation is enabled, this won't work because all the types
+      // are renamed
+      if (sentryElement.type.startsWith(_privateDelimiter) ||
+          (sentryElement.identifier?.startsWith(_privateDelimiter) ?? false)) {
+        privateElement = true;
+      } else {
+        parentSentryElement.children.add(sentryElement);
+      }
+
+      // we don't want to add private children but we still want to walk the tree
```

Comment:
Just out of curiosity: What would be a private children in Flutter?

Reply:
Dart has the concept of file private classes (similar to Swift). So a private child is a file private (widget) class, which is used to have better internal structure for widgets classes.